packages/liveblocks-react-ui/src/primitives/Markdown.tsx

```diff
@@ -1,25 +1,105 @@
+import { Slot } from "@radix-ui/react-slot";
 import { Lexer, type Tokens } from "marked";
 import {
-  type ComponentPropsWithoutRef,
   type ComponentType,
+  forwardRef,
   memo,
   type ReactNode,
   useMemo,
 } from "react";
 
-// TODO: Expose all Markdown tags in a holistic way
-//       (e.g. instead of `h1`, `h2`, etc, maybe `Heading` with a `level` prop? Maybe not every inline components should be exposed?)
+import type { ComponentPropsWithSlot } from "../types";
+
 export type MarkdownComponents = {
-  CodeBlock: ComponentType<{ language: string | null; code: string }>;
-  Anchor: ComponentType<ComponentPropsWithoutRef<"a">> | "a";
+  CodeBlock: ComponentType<MarkdownComponentsCodeBlockProps>;
+  Link: ComponentType<MarkdownComponentsLinkProps>;
+  Heading: ComponentType<MarkdownComponentsHeadingProps>;
+  Image: ComponentType<MarkdownComponentsImageProps>;
+  Blockquote: ComponentType<MarkdownComponentsBlockquoteProps>;
```

Comment:
I prefer the opinionated/"editorialized" slots like `Heading` with a `level` prop instead of `h1`, `h2`, `h3`, etc, especially with cases like `CodeBlock` which could mix `pre`/`code`, curious to hear other thoughts on this.